wlauto/common/android/device.py

```diff
@@ -376,12 +389,11 @@ def uninstall(self, package):
 
     def uninstall_executable(self, executable_name):
         """
-        Requires root access.
 
         Added in version 2.1.3.
 
         """
-        on_device_executable = self.path.join(self.binaries_directory, executable_name)
+        on_device_executable = self.get_binary_path(executable_name, search_system_binaries=False)
```

Comment:
`on_device_executable` may now be `None` this should be detected and a meaningful error return (other wise, this would result in something like "could not uninstall None").